Userland/DevTools/HackStudio/ProjectConfig.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ProjectConfig.h"
+#include <AK/NonnullOwnPtr.h>
+#include <LibCore/File.h>
+
+namespace HackStudio {
+
+ProjectConfig::ProjectConfig(JsonObject config)
+    : m_config(move(config))
+{
+}
+
+ErrorOr<NonnullOwnPtr<ProjectConfig>> ProjectConfig::try_load_projects_config(String path)
+{
+    auto file = TRY(Core::File::open(path, Core::OpenMode::ReadOnly));
+    auto file_contents = file->read_all();
+    file->close();
+
+    auto json = TRY(JsonValue::from_string(StringView { file_contents }));
+    if (!json.is_object())
+        return Error::from_string_literal("The topmost JSON element is not an object");
+
+    return adopt_own(*new ProjectConfig(json.as_object()));
+}
+
+NonnullOwnPtr<ProjectConfig> ProjectConfig::default_config()
```

Comment:
Since the default config doesn't contain any actual default values, why not call this function `ProjectConfig::create_empty()`?

Reply:
left overs of a previous implementation. fixed